picard/ui/infodialog.py

```diff
@@ -88,17 +102,77 @@ def _display_artwork_tab(self):
             if image.width and image.height:
                 infos.append(u"%d x %d" % (image.width, image.height))
             infos.append(image.mimetype)
-            item.setText(u"\n".join(infos))
-            self.ui.artwork_list.addItem(item)
+
+            img_wgt = self.artwork_table.get_coverart_widget(pixmap, "\n".join(infos))
+            self.artwork_table.setCellWidget(row, col, img_wgt)
+            self.artwork_table.setItem(row, col, item)
+            row += 1
+
+    def _display_artwork_type(self):
+        """Display image type in Type column.
+        If both existing covers and new covers are to be displayed, take union of both cover types list.
+        """
+        types = [image.types_as_string() for image in self.obj.metadata.images]
+        if self.display_existing_artwork:
+            existing_types = [image.types_as_string() for image in self.obj.orig_metadata.images]
+            #Merge both types and existing types list in sorted order.
+            #If a type is common in both types and existing_types list, append it only once.
+            #Otherwise append type in final list according to order.
+            temp = []
+            i = 0
+            j = 0
+            while i != len(types) and j != len(existing_types):
+                if types[i] > existing_types[j]:
+                    temp.append(existing_types[j])
+                    j += 1
+                elif types[i] < existing_types[j]:
+                    temp.append(types[i])
+                    i += 1
+                else:
+                    temp.append(types[i])
+                    i += 1
+                    j += 1
+            if i == len(types):
+                temp.extend(existing_types[j:])
+            else:
+                temp.extend(types[i:])
+            types = temp
```

Comment:
> But the lists you threw at it are unsorted, that's why.

Those types come from https://github.com/metabrainz/picard/blob/66ed358093ff4ae1c80b9fd33019390ae08b11cd/picard/coverart/image.py#L322 which doesn't sort types.
I'm perhaps missing something, but it seems to me there is no guarantee both lists share the same order.